compiler/ilgen/MethodBuilder.cpp

```diff
@@ -80,16 +86,22 @@ namespace OMR
 
 MethodBuilder::MethodBuilder(TR::TypeDictionary *types, OMR::VirtualMachineState *vmState)
    : TR::IlBuilder(asMethodBuilder(), types),
+   // Note: Any allocations in the following memory region that will get deallocated by C++ destruction semantics along with this MethodBuilder class, should instead
+   // be explicitly cleared / deallocated in the MethodBuilder destructor *before* the region itself is destroyed in order to avoid undefined behaviour from invalid
+   // memory references.
```

Comment:
The comment doesn't recommend deallocating the maps, since as you said, they are allocated within the MethodBuilder object. It recommends to deallocate "any allocations *in the region*", which in our case are the actual `std::pairs` stored by the map.

To avoid any ambiguity, I'm replacing the words "following memory region" with `_memoryRegion` and rephrasing the sentence overall to make it more explicit.

Also, my intention for putting the comment in the constructor was to make it more noticeable, since its a more likely place for a new contributor to allocate objects in the region. (I did add a note on the destructor though).